### PR TITLE
Docs/add doc to simulate data

### PR DESCRIPTION
**Pull request by signekb**
## Description

- This PR adds documentation (i.e., `roxygen2` comments) to each function in `simulate-data.R`. 
- This isn't technically necessary, but nice to help my understanding of what is going on in the functions created to simulate the data. 
- I have moved `add_fake_drug_name()` from the section "Simulate data", since I don't believe it fits there? However, I moved it to "Insert extra values to overrepresent certain values", which might not be a good location either. 
- I noticed that on L414, `dplyr::group_split()` is used which is `lifecycle: experimental`. Maybe we should use an alternative or is that fine?

### CODE REVIEW

**Review by Aastedet — APPROVED** [review]

File: data-raw/simulate-data.R [hunk 1]
```diff
@@ -47,13 +68,33 @@ create_fake_icd <- function(n, date = NULL) {
   }
 }
 
+#' Create ICD 10 diagnoses.
+
+#' @description
+#' ICD-8 is the 8th revision of the International Classification of Diseases.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:74]:
```suggestion
#' ICD-8 is the 8th revision of the International Classification of Diseases.
#' 
```

File: data-raw/simulate-data.R [hunk 2]
```diff
@@ -47,13 +68,33 @@ create_fake_icd <- function(n, date = NULL) {
...
+#' @param n The number of ICD-8 diagnoses to generate.
+#'
+#' @return n ICD-8 diagnoses as characters.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:77]:
```suggestion
#' @return A character vector of ICD-8 diagnoses.
```

File: data-raw/simulate-data.R [hunk 3]
```diff
@@ -29,12 +29,33 @@ library(rvest)
...
+#' pad_integers(x = c(1, 2, 3), width = 10)
 pad_integers <- function(x, width) {
   x |>
     stringr::str_trunc(width = width, side = "left", ellipsis = "") |>
     stringr::str_pad(width = width, side = "left", pad = "0")
 }
 
+#' Create ICD diagnoses.
+#'
+#' @param n The number of ICD-10 diagnoses to generate.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:51]:
```suggestion
#' @param n The number of ICD-8 or -10 diagnoses to generate.
```

File: data-raw/simulate-data.R [hunk 4]
```diff
@@ -29,12 +29,33 @@ library(rvest)
...
+#' @param date A date determining whether the diagnoses should be ICD-8 or ICD-10. If null, a random date will be sampled to determine which ICD revision the diagnosis should be from.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:52]:
```suggestion
#' @param date A date determining whether the diagnoses should be ICD-8 or ICD-10. If null, a random date will be sampled to determine which ICD revision the diagnosis should be from. In the Danish register, ICD-10 is used after 1994.
```

File: data-raw/simulate-data.R [hunk 5]
```diff
@@ -29,12 +29,33 @@ library(rvest)
...
+#'
+#' @return n ICD-10 diagnoses as characters.
+#'
+#' @examples
+#' create_fake_icd(10)
+#' #' create_fake_icd(5, "1995-04-19")
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:58]:
```suggestion
#' create_fake_icd(5, "1995-04-19")
```

File: data-raw/simulate-data.R [hunk 6]
```diff
@@ -47,13 +68,33 @@ create_fake_icd <- function(n, date = NULL) {
...
+#' @examples
+#' create_fake_icd8(1)
 create_fake_icd8 <- function(n) {
   here("data-raw/icd8-codes.csv") |>
     read_csv() |>
     pull(icd8) |>
     sample(size = n, replace = TRUE)
 }
 
+#' Create fake ICD 10 diagnoses.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:88]:
```suggestion
#' Create a vector of random ICD-10 diagnoses.
```

File: data-raw/simulate-data.R [hunk 7]
```diff
@@ -47,13 +68,33 @@ create_fake_icd <- function(n, date = NULL) {
...
+#'
+#' @description
+#' ICD-10 is the 10th revision of the International Classification of Diseases.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:91]:
```suggestion
#' ICD-10 is the 10th revision of the International Classification of Diseases.
#' 
```

File: data-raw/simulate-data.R [hunk 8]
```diff
@@ -47,13 +68,33 @@ create_fake_icd <- function(n, date = NULL) {
...
+#' @param n An integer determining how many diagnoses will be created.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:92]:
```suggestion
#' @param n An integer determining how many diagnoses to create.
```

File: data-raw/simulate-data.R [hunk 9]
```diff
@@ -47,13 +68,33 @@ create_fake_icd <- function(n, date = NULL) {
...
+#'
+#' @return n ICD-10 diagnoses as characters.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:94]:
```suggestion
#' @return A character vector of ICD-10 diagnoses.
```

File: data-raw/simulate-data.R [hunk 10]
```diff
@@ -62,6 +103,17 @@ create_fake_icd10 <- function(n) {
     sample(size = n, replace = TRUE)
 }
 
+#' Create fake ATC codes.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:106]:
```suggestion
#' Create a vector with random ATC codes
```

File: data-raw/simulate-data.R [hunk 11]
```diff
@@ -62,6 +103,17 @@ create_fake_icd10 <- function(n) {
...
+#'
+#' @description
+#' Anatomical Therapeutic Chemical (ATC) codes are unique medicine codes
+#' based on on what organ or system it works on and how it works.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:110]:
```suggestion
#' based on on what organ or system it works on and how it works.
#' 
```

File: data-raw/simulate-data.R [hunk 12]
```diff
@@ -62,6 +103,17 @@ create_fake_icd10 <- function(n) {
...
+#' @param n The number of fake ATC codes to generate.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:111]:
```suggestion
#' @param n The number of random ATC codes to generate.
```

File: data-raw/simulate-data.R [hunk 13]
```diff
@@ -62,6 +103,17 @@ create_fake_icd10 <- function(n) {
...
+#'
+#' @return n ATC codes as characters.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:113]:
```suggestion
#' @return A character vector of ATC codes.
```

File: data-raw/simulate-data.R [hunk 14]
```diff
@@ -70,24 +122,63 @@ create_fake_atc <- function(n) {
     sample(n, replace = TRUE)
 }
 
+#' Create fake dates.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:125]:
```suggestion
#' Create fake dates
```

File: data-raw/simulate-data.R [hunk 15]
```diff
@@ -70,24 +122,63 @@ create_fake_atc <- function(n) {
...
+#'
+#' @param n The number of dates to generate.
+#' @param from A date determining the first date in the interval to sample from.
+#' @param to A date determining the last date in the interval to sample from.
+#'
+#' @return n dates.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:131]:
```suggestion
#' @return A vector of dates.
```

File: data-raw/simulate-data.R [hunk 16]
```diff
@@ -70,24 +122,63 @@ create_fake_atc <- function(n) {
...
+#'
+#' @examples
+#' create_fake_date(20)
+#' create_fake_date(20, "1995-04-19", "2024-04-19")
 create_fake_date <- function(n, from = "1977-01-01", to = lubridate::today()) {
   seq(as_date(from), as_date(to), by = "day") |>
     sample(n, replace = TRUE)
 }
 
+#' Create padded integers.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:141]:
```suggestion
#' Create a vector of random zero-padded integers.
```

File: data-raw/simulate-data.R [hunk 17]
```diff
@@ -70,24 +122,63 @@ create_fake_atc <- function(n) {
...
 }
 
+#' Create fake NPU codes.
+#'
+#' @description
+#' Nomenclature for Properties and Units (NPUs) are codes that identifies
+#' laboratory results.
+#' @param n The number of NPUs to create.
+#'
+#' @return n NPUs as characters.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:163]:
```suggestion
#' @return A character vector.
```

File: data-raw/simulate-data.R [hunk 18]
```diff
@@ -70,24 +122,63 @@ create_fake_atc <- function(n) {
...
+#' @examples
+#' create_fake_npu(4)
 create_fake_npu <- function(n) {
   stringr::str_c(
     "NPU",
     create_padded_integer(n, 5)
   )
 }
 
+#' Create department specialties.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:174]:
```suggestion
#' Create a vector of random department specialties
```

File: data-raw/simulate-data.R [hunk 19]
```diff
@@ -70,24 +122,63 @@ create_fake_atc <- function(n) {
...
+#'
+#' @param n The number of department specialties to create.
+#'
+#' @return n random department specialties.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:178]:
```suggestion
#' @return A character vector.
```

File: data-raw/simulate-data.R [hunk 20]
```diff
@@ -97,6 +188,15 @@ create_fake_hovedspeciale_ans <- function(n) {
     sample(n, replace = TRUE)
 }
 
+#' Create drug names from ATC codes.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:191]:
```suggestion
#' Create a vector of drug names based on a vector of ATC codes
```

File: data-raw/simulate-data.R [hunk 21]
```diff
@@ -97,6 +188,15 @@ create_fake_hovedspeciale_ans <- function(n) {
...
+#'
+#' @param atc A character describing an ATC code.
+#'
+#' @return A character with the drug name of the given ATC code.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:195]:
```suggestion
#' @return A character vector with the drug name of the given ATC code.
```

File: data-raw/simulate-data.R [hunk 22]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
     sample(length(atc), replace = TRUE)
 }
 
+#' Transform date(s) to the format wwyy.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:208]:
```suggestion
#' Transform date(s) to the format wwyy
```

File: data-raw/simulate-data.R [hunk 23]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
...
+#'
+#' @param x A date or a vector of dates.
+#'
+#' @return The input date(s) in the format wwyy.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:212]:
```suggestion
#' @return A vector of dates in the format wwyy.
```

File: data-raw/simulate-data.R [hunk 24]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
...
+#'
+#' @examples
+#' to_wwyy("2020-12-01")
+#' to_wwyy(c("2020-01-12", "1995-04-19"))
 to_wwyy <- function(x) {
   paste0(lubridate::isoweek(lubridate::as_date(x)), stringr::str_sub(lubridate::isoyear(lubridate::as_date(x)), -2))
 }
 
+#' Transform date(s) to the format yyyymmdd.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:221]:
```suggestion
#' Transform date(s) to the format yyyymmdd
```

File: data-raw/simulate-data.R [hunk 25]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
...
+#'
+#' @param x A date or a vector of dates.
+#'
+#' @return The input date(s) in the format yyyymmdd.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:225]:
```suggestion
#' @return A vector of dates in the format yyyymmdd.
```

File: data-raw/simulate-data.R [hunk 26]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
...
   format(lubridate::as_date(x), format = "%Y%m%d")
 }
 
 # Insert extra values to overrepresent certain values ------------------------------------------------------
 
+#' Generate logic based on a probability
+#'
+#' @param proportion A double between 0 and 1.
+#'
+#' @return A logic. TRUE if the random number is less than the proportion,
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:240]:
```suggestion
#' @return A logic vector. TRUE if the random number is less than the proportion,
```

File: data-raw/simulate-data.R [hunk 27]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
...
+#' otherwise FALSE.
+#'
+#' @examples
+#' insertion_rate(0.3)
 insertion_rate <- function(proportion) {
   runif(1) < proportion
 }
 
+#' Insert ATC based on a proportion
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:249]:
```suggestion
#' Insert specific ATC codes based on a proportion
```

File: data-raw/simulate-data.R [hunk 28]
```diff
@@ -105,20 +205,56 @@ create_fake_drug_name <- function(atc) {
...
+#'
+#' @param data A tibble
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:251]:
```suggestion
#' @param data A tibble.
```

File: data-raw/simulate-data.R [hunk 29]
```diff
@@ -149,8 +285,17 @@ insert_specific_atc <- function(data, proportion = 0.3) {
...
 
-# Insert a few cases where purchases of metformin are used for other purposes
-# than diabetes.
+#' Insert cases where metformin is used for other purposes than diabetes
+#'
+#' @description
+#' This function uses the variable 'indo' which is the code for the underlying
+#' condition treated by the prescribed medication.
+#'
+#' @param data A tibble
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:294]:
```suggestion
#' @param data A tibble.
```

File: data-raw/simulate-data.R [hunk 30]
```diff
@@ -170,7 +315,14 @@ insert_false_metformin <- function(data, proportion = 0.05) {
     )
 }
 
-# Insert some false positives for Wegovy and Saxenda.
+# Insert false positives for Wegovy and Saxenda.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:318]:
```suggestion
# Insert false positives for Wegovy and Saxenda
```

File: data-raw/simulate-data.R [hunk 31]
```diff
@@ -185,6 +337,13 @@ insert_false_drug_names <- function(data, proportion = 0.05) {
     )
 }
 
+#' Insert additional analysis codes for HbA1c.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:340]:
```suggestion
#' Insert additional analysis codes for HbA1c
```

File: data-raw/simulate-data.R [hunk 32]
```diff
@@ -201,11 +360,32 @@ insert_analysiscode <- function(data, proportion = 0.3) {
     )
 }
 
+#' Add drug names (from ATC codes).
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:363]:
```suggestion
#' Add drug names (from ATC codes)
```

File: data-raw/simulate-data.R [hunk 33]
```diff
@@ -201,11 +360,32 @@ insert_analysiscode <- function(data, proportion = 0.3) {
...
+      )
+    )
+}
+
 # TODO: Need a function to reuse recnum and dw_ek_kontakt in LPR data
 
 # Simulate data -----------------------------------------------------------
 
-# use the simulation definition data to simulate some data
+#' Simulate data based on simulation definitions.
```
Comment (lwjohnst86) [inline data-raw/simulate-data.R:383]:
```suggestion
#' Simulate data based on simulation definitions
```

**Review by lwjohnst86 — CHANGES_REQUESTED** [review]
Just some small edits/clarifications ☺️

**Review by lwjohnst86 — APPROVED** [review]

**Comment by lwjohnst86** [comment]
Related to your question about `group_split()`. That function has been in experimental state for the last 4 years, so I don't expect it to disappear anytime soon. If these functions were critical to the package, I would change it, but it isn't critical.

**Pull request merged**